### ISSUE TITLE
AIR and AIR_PURGE Triggers Are Opposite

### ISSUE THREAD

**Issue by graysonk95** [RigsOfRods/rigs-of-rods#445]
In the original Soundscript documentation it says the two triggers should behave as follows:
*air* : *active* *air* *suspension* *activation*
*air_purge* : *compressed* *air* *purge*
http://www.rigsofrods.com/wiki/pages/Soundscript_files

The two triggers are currently reversed in their behavior.
"Air_purge" should be the air dryer, whereas "air" should be triggered by the depreciated shocks2 l/r flags or the depreciated wheels2 pressure adjustment feature (Issue https://github.com/RigsOfRods/rigs-of-rods/issues/446).

As it stands AIR_PURGE plays with the air suspension (l/r flags) and AIR plays as the truck's air dryer.

**Comment by only-a-ptr** [comment]
I'm researching how the triggers work in code. Work in progress.

Sound triggers
==============

definitions and parsing
-----------------------

SS_TRIG_AIR: https://github.com/only-a-ptr/rigs-of-rods/blob/editor-python-dirty/source/main/audio/SoundScriptManager.h#L50
> Trigger source: air: https://github.com/only-a-ptr/rigs-of-rods/blob/editor-python-dirty/source/main/audio/SoundScriptManager.cpp#L533

SS_TRIG_AIR_PURGE: https://github.com/only-a-ptr/rigs-of-rods/blob/editor-python-dirty/source/main/audio/SoundScriptManager.h#L60
> Trigger source: air_purge: https://github.com/only-a-ptr/rigs-of-rods/blob/editor-python-dirty/source/main/audio/SoundScriptManager.cpp#L543

no error here.

Usage of SS_TRIG_AIR in code.
-----------------------------

RoRFrameListener.cpp:
> Pressure control logic: https://github.com/only-a-ptr/rigs-of-rods/blob/editor-python-dirty/source/main/gameplay/RoRFrameListener.cpp#L749    
   >> EV_COMMON_PRESSURE_LESS >> trigStart(SS_TRIG_AIR);
   >> EV_COMMON_PRESSURE_MORE >> trigStart(SS_TRIG_AIR);
  >> none pressed >> trigStop(SS_TRIG_AIR);

MainThread.cpp:
>    OnChangedVehicle() https://github.com/only-a-ptr/rigs-of-rods/blob/editor-python-dirty/source/main/main_sim/MainThread.cpp#L1284
>>        trigStop(previous_vehicle, SS_TRIG_AIR); https://github.com/only-a-ptr/rigs-of-rods/blob/editor-python-dirty/source/main/main_sim/MainThread.cpp#L1357
        
BeamForcesEuler.cpp:
> CalcShocks() + stability code 
> https://github.com/only-a-ptr/rigs-of-rods/blob/editor-python-dirty/source/main/physics/BeamForcesEuler.cpp#L1077
>>    trigStart/trigStop(SS_TRIG_AIR);  
>>  https://github.com/only-a-ptr/rigs-of-rods/blob/editor-python-dirty/source/main/physics/BeamForcesEuler.cpp#L1129
        
**NOTE: There's a bug in spawning code (damn copy-pasting!)** :blush: 
>    RigSpawner::ProcessShock(RigDef::Shock & def) 
>    https://github.com/only-a-ptr/rigs-of-rods/blob/editor-python-dirty/source/main/physics/input_output/RigSpawner.cpp#L4158
    
   >> Left active shock: https://github.com/only-a-ptr/rigs-of-rods/blob/editor-python-dirty/source/main/physics/input_output/RigSpawner.cpp#L4179
 >>   Right active shock: https://github.com/only-a-ptr/rigs-of-rods/blob/editor-python-dirty/source/main/physics/input_output/RigSpawner.cpp#L4185

**Comment by graysonk95** [comment]
Neat, I didn't know that air pressure was something that was calculated realistically. With no type of air gauge, the sound always seemed random to me.
It makes me wonder why air command beams, actual air brakes, air suspension, or even just an air tank gauge were never implemented... Correct me if I'm wrong, but the all of the code for the "apressure" calculation seems to be there just for the air_purge sound?
Is apressure used anywhere else?

Also, I just did a test where I replaced the sounds with voice recordings and lo and behold they are correct.
I was mistaken in thinking they were switched. This ticket can be closed.
